src/p2p/net_node.h

```diff
@@ -117,6 +117,18 @@ namespace nodetool
         support_flags(0),
         m_in_timedsync(false)
     {}
+    p2p_connection_context_t(const p2p_connection_context_t &context):
+      base_type(context),
+      fluff_txs{},
+      flush_time{},
+      peer_id(context.peer_id),
+      support_flags(context.support_flags),
+      m_in_timedsync(context.m_in_timedsync),
+      sent_addresses{}
+    {}
+    p2p_connection_context_t &operator=(const p2p_connection_context_t &context) {
+      return *new(this) p2p_connection_context_t(context);
```

Comment:
Is it still memory leak with added `this->~p2p_connection_context_t` ?
Why is it undefined behaviour ?
What's about this place ?
https://github.com/monero-project/monero/blob/82149bfe441e56d5f9d72a13ffa25ce1df6d8391/contrib/epee/include/net/net_utils_base.h#L431

Reply:
It's undefined behaviour because it doesn't destruct any members in the current object, it just rams new initialization of top of the existing members.  So the vector and the set, for instance, are never destructed and their allocated memory is never freed.

An explicit constructor would indeed explicitly destruct, and would therefore solve the UB.  But you still *shouldn't* do either because it is still a horrible abuse of the purpose of placement new (and it is horrible where it is being done in epee, too).

Reply:
Yeah if the constructor throws on line 131, then the object is in UB zombie state.

I'm also not a fan of using the copy-constructor and copy-assignment like this. Its not actually copying the source object. Just create a new function and call it on the single line (IIRC) that needs this functionality.

Reply:
The other reason for a new function is that the base class might have the same problem at some point. So some properly named function would be useful.